.github/workflows/_rust_lints.yml

```diff
@@ -45,6 +47,20 @@ jobs:
             exit 1    # Fail the workflow
           fi
 
+  examples:
+    if: (!cancelled() && inputs.isRustExample)
+    runs-on: [self-hosted]
+    steps:
+      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+      - name: Check examples
+        run: |
+          for manifest in ./**/examples/**/Cargo.toml; do
+              echo "Checking format of $manifest"
+              cargo +nightly ci-fmt --manifest-path $manifest
+              echo "Clippy linting $manifest"
+              cargo clippy --manifest-path $manifest --all-features
```

Comment:
Should probably have these as well: udeps, typos, deny, dprint-fmt, license check